app/models/location.rb

```diff
@@ -84,19 +102,23 @@ class Location < ApplicationRecord
 
   scope :active_status, lambda {
     where.not(id: joins(:location_deactivation_periods).reactivating_soon.pluck(:id))
-    .where.not(id: joins(scheme: [:owning_organisation]).deactivated_by_organisation.pluck(:id))
-    .where.not(id: joins(:location_deactivation_periods).deactivated_directly.pluck(:id))
-    .where.not(id: incomplete.pluck(:id))
-    .where.not(id: joins(:location_deactivation_periods).deactivating_soon.pluck(:id))
-    .where.not(id: activating_soon.pluck(:id))
+         .where.not(id: joins(scheme: [:scheme_deactivation_periods]).reactivating_soon_by_scheme.pluck(:id))
+         .where.not(id: joins(:location_deactivation_periods).merge(Location.deactivated_directly).pluck(:id))
+         .where.not(id: joins(scheme: [:scheme_deactivation_periods]).merge(Location.deactivated_by_scheme).pluck(:id))
+         .where.not(id: joins(scheme: [:owning_organisation]).merge(Location.deactivated_by_organisation).pluck(:id))
+         .where.not(id: incomplete.pluck(:id))
+         .where.not(id: joins(:location_deactivation_periods).merge(Location.deactivating_soon_directly).pluck(:id))
+         .where.not(id: joins(scheme: %i[owning_organisation scheme_deactivation_periods]).merge(Location.deactivating_soon_by_scheme).pluck(:id))
+         .where.not(id: activating_soon.pluck(:id))
   }
 
   scope :active, lambda { |date = Time.zone.now|
-    where.not(id: joins(:location_deactivation_periods).reactivating_soon(date).pluck(:id))
-         .where.not(id: joins(scheme: [:owning_organisation]).deactivated_by_organisation.pluck(:id))
-         .where.not(id: joins(:location_deactivation_periods).deactivated_directly(date).pluck(:id))
+    where.not(id: joins(scheme: [:scheme_deactivation_periods]).reactivating_soon_by_scheme.pluck(:id))
```

Comment:
Would `.where(scheme: Scheme.active(date))` already remove anything that's `reactivating_soon_by_scheme` and `deactivated_by_organisation` ?

Reply:
Yes I think so, will remove them.